lib/fuel-rest.js

```diff
@@ -84,16 +84,18 @@ class FuelRest {
 				return tokenInfo;
 			})
 			.then(tokenInfo => {
-				let retry = false;
+				let retry = options.retry || false;
 				const consolidatedOpts = {};
 				const authOptions = clone(options.auth);
+				const headers = options.headers;
 
 				options.uri = helpers.resolveUri(this.origin, options.uri);
-				options.headers = Object.assign({}, this.defaultHeaders, options.headers);
+				options.headers = Object.assign({}, this.defaultHeaders, options.headers, {
+					Authorization: 'Bearer ' + tokenInfo.accessToken
+				});
 
-				if (!options.headers.Authorization) {
-					options.headers.Authorization = 'Bearer ' + tokenInfo.accessToken;
-					retry = options.retry || false;
+				if(headers && headers.Authorization) {
+					retry = false;
```

Comment:
Just a little worries about the new logic here.
This line means no matter what the value of options.retry is, retry is alway false as long as Authorization header is already set.

So if someone calls the functions with options that has the header already, it won't retry even options.retry === true.

You can test this by adding the following code to the unit test you modified:
```
requestOptions.headers = {
			Authorization: "Bear some_token"
		};
```

If not a big problem but worth to have a think about it.

Reply:
I'm feeling like this will be okay just to omit this check at all. 

I believe the only reason that it was there in the first place is because the retried request would have the Authorization header in it. When I think about it now, it (I) should've been done differently from the start. 

If a person makes an API request, it should try with everything they passed. If it fails, forcing a retry, we should replace the token no matter what to try again with "something we know should be valid."

Reply:
The latest commit should solve this